### PR TITLE
fix duplicate display index after using duplicate_entry_button

### PR DESCRIPTION
**Pull request by steve02081504**
## Checklist:

- [x] I have read the [Contributing guidelines](https://github.com/SillyTavern/SillyTavern/blob/release/CONTRIBUTING.md).

### CODE REVIEW

File: public/scripts/world-info.js [hunk 1]
```diff
@@ -3220,6 +3220,10 @@ export function duplicateWorldInfoEntry(data, uid) {
     const entry = createWorldInfoEntry(data.name, data);
     Object.assign(entry, originalData);
 
+    // Update Display indexs
+    data.entries.filter(x => x.displayIndex > entry.displayIndex).forEach(x => x.displayIndex++);
```
Comment (Cohee1207) [inline public/scripts/world-info.js]: You also need to call `setWIOriginalDataValue(data, uid, 'extensions.display_index', item.displayIndex);` for the entry if you change any of its data.

**Review by Cohee1207 — CHANGES_REQUESTED** [review]

File: public/scripts/world-info.js (hunk 1 above)
Comment (steve02081504) [reply public/scripts/world-info.js]: fixed in https://github.com/SillyTavern/SillyTavern/commit/498051a8f79b04509a95a90163430f1baa34ac92

**Comment by Wolfsblvt** [comment]
I've never ran into issues with this. There is no problem with having duplicate displayIndex values. It's just a primary sorting if you want it to, before the uid sorting gets done.

If you never chose "Custom" as a sorting options, displayIndex will even be null.
So your code would also need to take in mind that existing displayIndex values could be null, for the duplication.

**Comment by Cohee1207** [comment]
> I've never ran into issues with this. There is no problem with having duplicate displayIndex values. It's just a primary sorting if you want it to, before the uid sorting gets done.

Yeah, true. This fixes what's not broken in the ST itself, since nothing but UID is required to be unique.

**Pull request closed without merging**